tests/lava/magma/runtime/test_ref_var_ports.py

```diff
@@ -220,6 +266,57 @@ def test_implicit_Ref_Var_port_read(self):
             np.all(recv.var3.get() == np.array([17., 17.])))
         recv.stop()
 
+    def test_multiple_var_ports(self):
+        """Tests connecting multiple RefPorts to different Vars of a target
+        Process. The RefPort sends data after the first time step to the
+        VarPort, starting with (5 + current time step) = 7). After 2 time steps
+        the value for var2 and var4 is expected to be 7."""
+
+        sender1 = P1()
+        sender2 = P1()
+
+        recv = P2()
+
+        sender1.ref1.connect_var(recv.var2)
+        sender2.ref3.connect_var(recv.var4)
+
+        simple_sync_domain = SyncDomain("simple", LoihiProtocol(),
+                                        [sender1, sender2, recv])
+
+        # Run for two time steps
+        recv.run(RunSteps(num_steps=2, blocking=True),
+                 MyRunCfg(custom_sync_domains=[simple_sync_domain]))
+
+        self.assertTrue(
+            np.all(recv.var2.get() == np.array([7., 7., 7.])))
+        self.assertTrue(
+            np.all(recv.var4.get() == np.array([7., 7.])))
+        recv.stop()
+
+    def test_hierarchical_ref_ports(self):
+        """Tests if sending data via a RefPort of an instance of the
+        hierarchical process HP1 to a Var of an instance of the hierarchical
+        process HP2 works. The RefPort of HP1 mirrors the RefPort ref1
+        of its nested process P1. HP2 has a Var h_var which aliases the Var var2
```

Comment:
I'm not sure this is correct. Isn't it that the data is sent from `p1.ref1` to `hp1.h_ref` and from there directly into `hp2.h_var`? The data ends up both in `hp2.h_var` and `p2.var2` but there is no direct communication between `p1` and `p2`. It seems to me that for this test, you don't need HP2. You could just connect `hp1.h_ref` directly to `p2.var2` if you make `p2` the receiver: `recv = P2()`

Reply:
I have rewritten the docstring.

Yes, one could also connect h_ref directly to var2, but I wanted to show that hierarchical processes can be also be connected with RefPorts. I also do not need HP1, I could just connect p1.ref1 to p2.var2, but this is not the point.